### PR TITLE
tp-qemu: Add some subtests when balloon driver in use.

### PR DESCRIPTION
**Pull request by CongSmile**
Signed-off-by: Cong Li <coli@redhat.com>

Based on: autotest/tp-qemu#463

id: 1230672 1231010 1231013 1231655

### CODE REVIEW

File: generic/tests/cfg/balloon_in_use.cfg [hunk 1]
```diff
@@ -0,0 +1,30 @@
+- balloon_in_use:
+    type = driver_in_use
+    only Windows
+    extra_params += " -balloon virtio"
```
Comment (vipmike007) [inline generic/tests/cfg/balloon_in_use.cfg]: it is the RHEL5 cli ,can you replace it w/ RHEL6/7 one ?

Comment (CongSmile) [inline generic/tests/cfg/balloon_in_use.cfg]: ah, yes, it's RHEL.5 CML. I will update it.

Thanks Mike.

**Comment by CongSmile** [comment]
Hi @vipmike007 

I have updated the code, could you help review it again?

Thanks.

File: qemu/tests/cfg/balloon_in_use.cfg [hunk 2]
```diff
@@ -0,0 +1,30 @@
+- balloon_in_use:
+    type = driver_in_use
+    only Windows
+    extra_params += "-device virtio-balloon-pci,id=ballooning"
+    iterations = 5
```
Comment (xutian) [inline qemu/tests/cfg/balloon_in_use.cfg]: why need test 5 times?

Comment (CongSmile) [inline qemu/tests/cfg/balloon_in_use.cfg]: According to the request, needs a balloon loop for the test.

Comment (xutian) [inline qemu/tests/cfg/balloon_in_use.cfg]: got it, thanks.

File: qemu/tests/cfg/balloon_in_use.cfg [hunk 3]
```diff
@@ -0,0 +1,30 @@
+- balloon_in_use:
+    type = driver_in_use
+    only Windows
+    extra_params += "-device virtio-balloon-pci,id=ballooning"
```
Comment (xutian) [inline qemu/tests/cfg/balloon_in_use.cfg]: framework has support balloon params to add balloon device, so please use it to setup your balloon device

File: qemu/tests/cfg/balloon_in_use.cfg [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    iterations = 5
+    ratio = 0.5
```
Comment (suqinhuang) [inline qemu/tests/cfg/balloon_in_use.cfg]: Hi coli,
Could you double check value of ratio, it's a little big

File: qemu/tests/cfg/balloon_in_use.cfg (hunk 2 above)
Comment (suqinhuang) [inline qemu/tests/cfg/balloon_in_use.cfg]: Hi coli,
Didn't find the repeating request

File: qemu/tests/cfg/balloon_in_use.cfg [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    login_timeout = 800
+    start_vm = yes
+    kill_vm_on_error = yes
+    driver_name = "balloon"
+    run_bgstress = balloon_check
```
Comment (suqinhuang) [inline qemu/tests/cfg/balloon_in_use.cfg]: balloon_check is not the same as the stress from the requirement, is it acceptable?

File: qemu/tests/cfg/balloon_in_use.cfg (hunk 4 above)
Comment (vipmike007) [inline qemu/tests/cfg/balloon_in_use.cfg]: We need to consider windows 32 bit guests ,  as well 
if we start vm  w/ 8GB ,then balloon it to 4G on qemu side ,there should be no empty mem pages on windows guests.

**Comment by CongSmile** [comment]
Based on #447

File: qemu/tests/cfg/balloon_in_use.cfg [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+    driver_name = "balloon"
+    run_bgstress = balloon_stress
+    bg_stress_run_flag = balloon_test
+    free_mem_cmd = cat /proc/meminfo |grep MemFree
+    video_url = http://FILESHARE.COM/pub/section2/kvmauto/video/big_buck_bunny_480p_stereo.avi
+    x86_64:
+        program_files = "%ProgramFiles(x86)%"
+    i386:
+        program_files = "%ProgramFiles%"
+    Windows:
```
Comment (spcui) [inline qemu/tests/cfg/balloon_in_use.cfg]: pls move these lines to guest-os cfg

File: qemu/tests/cfg/balloon_in_use.cfg [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+    iterations = 5
+    ratio = 0.5
+    login_timeout = 800
+    start_vm = yes
+    kill_vm_on_error = yes
+    check_guest_bsod = yes
+    driver_name = "balloon"
+    run_bgstress = balloon_stress
+    bg_stress_run_flag = balloon_test
+    free_mem_cmd = cat /proc/meminfo |grep MemFree
```
Comment (xiagao) [inline qemu/tests/cfg/balloon_in_use.cfg]: Hi, coli
Seems "free_mem_cmd" is not useful for windows guest. Could you pls check it？

File: qemu/tests/cfg/balloon_in_use.cfg (hunk 6 above)
Comment (xiagao) [inline qemu/tests/cfg/balloon_in_use.cfg]: Hi coli,
As we talked ,could you check if it's necessary to figure out  "windows"  for the following parameters?

thanks,
xiagao

File: qemu/tests/cfg/balloon_in_use.cfg (hunk 4 above)
Comment (CongSmile) [inline qemu/tests/cfg/balloon_in_use.cfg]: Hi @suqinhuang 

Have confirmed with the feature owner,the ratio is fine .

**Comment by CongSmile** [comment]
Hi @suqinhuang and @xutian 

Could you help review this PR again?

Thanks.

**Comment by suqinhuang** [comment]
Acked-by: Suqin Huang

**Comment by suqinhuang** [comment]
@CongSmile could you provide the patch for  guest memory limit

**Comment by CongSmile** [comment]
Hi @suqinhuang 

As talked with @xutian , the memory limit will be done in the balloon class.
For this case, no need to update the memory limit, just only need update case "balloon_stress".

Thanks.

**Comment by suqinhuang** [comment]
Acked-by: Suqin Huang

**Comment by suqinhuang** [comment]
10:37:15 test             L0518 ERROR| Traceback (most recent call last):

10:37:15 test             L0518 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_vt/test.py", line 414, in _run_avocado
    raise test_exception

10:37:15 test             L0518 ERROR| VMDeadError: VM is dead

10:37:15 test             L0535 ERROR| ERROR smp_1.4096m.repeat1.run_test.Host_RHEL.m7.u2.qcow2.virtio_blk.up.virtio_net.Win8.x86_64.1.io-github-autotest-qemu.balloon_in_use.with_shutdown.before_bg_test -> VMDeadError: VM is dead

**Comment by suqinhuang** [comment]
Other cases works well

**Comment by maxujun** [comment]
I met the same problem with guest win7
ERROR smp_2.2048m.repeat1.run_test.Host_RHEL.7.2.qcow2.virtio_scsi.up.virtio_net.Win7.x86_64.sp1.io-github-autotest-qemu.balloon_in_use.with_shutdown.before_bg_test -> VMDeadError: VM is dead    detail: "WARNING: aio=native was specified for '/usr/share/avocado/data/avocado-vt/isos/windows/winutils.iso', but it requires cache.direct=on, which was not specified. Falling back to aio=threads.\n         This will become an error condition in future QEMU versions.\nehci warning: guest updated active QH\n"

**Comment by CongSmile** [comment]
There should be no problem now

15:54:53 INFO | PASS smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u1.qcow2.virtio_blk.up.virtio_net.Win8.x86_64.1.io-github-autotest-qemu.balloon_in_use.with_shutdown.before_bg_test

Please check https://github.com/autotest/tp-qemu/commit/1734bc590f1531bd88ed9ddb93d51e280597131b .

Thanks.

**Comment by suqinhuang** [comment]
Acked-by: Suqin Huang

**Comment by maxujun** [comment]
Test passed
Acked-by: Xujun Ma

**Comment by xutian** [comment]
looks ok for me, merged it. thanks @CongSmile , @maxujun , @suqinhuang and @spcui

**Pull request merged**